test/libdnf/rpm/solv/test_solv_map.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "test_solv_map.hpp"
+
+#include <cstdint>
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION(SolvMapTest);
+
+
+void SolvMapTest::setUp() {
+    map1 = new libdnf::rpm::solv::SolvMap(32);
+    map1->add(0);
+    map1->add(2);
+    map1->add(28);
+    map1->add(30);
+
+    map2 = new libdnf::rpm::solv::SolvMap(32);
+    map2->add(0);
+    map2->add(1);
+}
+
+
+void SolvMapTest::tearDown() {
+    delete map1;
+    delete map2;
+}
+
+
+void SolvMapTest::test_add() {
+    // test if the first byte in the bitmaps contains expected values
+    CPPUNIT_ASSERT(*(map1->get_map()->map) == 5);
+    CPPUNIT_ASSERT(*(map2->get_map()->map) == 3);
+
+    // test invalid ranges
+    CPPUNIT_ASSERT_THROW(map1->add(-1), std::out_of_range);
+    CPPUNIT_ASSERT_THROW(map1->add(33), std::out_of_range);
+}
+
+
+void SolvMapTest::test_contains() {
+    CPPUNIT_ASSERT(map1->contains(0) == true);
+    CPPUNIT_ASSERT(map1->contains(1) == false);
+
+    CPPUNIT_ASSERT(map2->contains(0) == true);
+    CPPUNIT_ASSERT(map2->contains(1) == true);
+
+    // test invalid ranges
+    CPPUNIT_ASSERT(map2->contains(-1) == false);
+    CPPUNIT_ASSERT(map2->contains(33) == false);
+}
+
+
+void SolvMapTest::test_remove() {
+    map1->remove(0);
+    map2->remove(0);
+    CPPUNIT_ASSERT(*(map1->get_map()->map) == 4);
+    CPPUNIT_ASSERT(*(map2->get_map()->map) == 2);
+
+    // test invalid ranges
+    CPPUNIT_ASSERT_THROW(map1->remove(-1), std::out_of_range);
+    CPPUNIT_ASSERT_THROW(map1->remove(33), std::out_of_range);
+}
+
+
+void SolvMapTest::test_union() {
+    *map2 |= *map1;
+    CPPUNIT_ASSERT(*(map2->get_map()->map) == 7);
```

Comment:
A think that test using of `map2->get_map()->map` is not transparent. How the result change when I will add id 30 into map1?
Also there is no test for initial state before tested operation is applied.

Reply:
ok, I'll use contains() instead.